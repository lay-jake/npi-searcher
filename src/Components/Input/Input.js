import React, { useEffect } from 'react'
import { useState } from 'react';
import { Col} from 'react-bootstrap';

export const Input = ({searchType,display,updateSearchField}) => {

  const handleDefault = () =>{
    switch(searchType){
         case "firstName":
          return "Exact or first two wildcards"
         case "lastName":
          return "Exact or first two wildcards"
         case "npiNumber":
          return "Exactly 10 digits"
         case "taxonomyDescription":
          return "Exact Description or Specialty"
         case "city":
          return "Exact or first two wildcards"
         case "state":
          return "Abrv. and Other Fields"
         case "zip":
          return "Exact Postal Code"
         default:
          return ""
    }
 }

const [inputText, setInputText] = useState("");
const [placeholder, setPlaceholder] = useState(handleDefault());
//UseEffect checks to see if inputText state has changed, if it does it sends the update to the store.

useEffect(()=>{
  updateSearchField({type:searchType,value:inputText})  
},[inputText])

const handleChange = e => {
    setInputText(e.target.value);
    };  

return (
      <Col xs={8} sm={"auto"} data-testid="input-element-1">
        <form data-testid="input-element-3">
            <label data-testid="input-element-2">
                {`${display}`}
                <input type='text' placeholder={placeholder} value={inputText} onChange = {handleChange}/>
            </label>
        </form>
      </Col>
   
  )
}
