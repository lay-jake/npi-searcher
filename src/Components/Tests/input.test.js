import {render, screen, cleanup} from '@testing-library/react'
import { Input } from '../Input/Input';
import renderer from 'react-test-renderer'


afterEach(()=>{
    cleanup();
})

test('Should Render the Input Component',()=>{
    const props ={searchtype:"InputTest",display:"This is a display field: ", updateSearchField: jest.fn()};
    render( <Input searchType={props.searchtype} display={props.display} updateSearchField={props.updateSearchField}/>)
    const inputElement = screen.getByTestId('input-element-1');
    expect (inputElement).toBeInTheDocument();
})

test('Should Render the correct display for the searchtype',()=>{
    const props ={searchtype:"InputTest",display:"This is a display field: ", updateSearchField: jest.fn()};
    render( <Input searchType={props.searchtype} display={props.display} updateSearchField={props.updateSearchField}/>)
    const inputElement = screen.getByTestId('input-element-2');
    expect (inputElement).toHaveTextContent('This is a display field');
})

test('Match the snapshot',()=>{
    const props = {searchtype:"InputTest",display:"This is a display field: ", updateSearchField: jest.fn()};
    const inputTree = renderer.create(<Input searchType={props.searchtype} display={props.display} updateSearchField={props.updateSearchField}/>).toJSON();
    expect(inputTree).toMatchSnapshot();
})


