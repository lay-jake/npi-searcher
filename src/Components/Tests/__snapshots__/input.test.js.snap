// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Match the snapshot 1`] = `
<div
  className="col-sm-auto col-8"
  data-testid="input-element-1"
>
  <form
    data-testid="input-element-3"
  >
    <label
      data-testid="input-element-2"
    >
      This is a display field: 
      <input
        onChange={[Function]}
        placeholder=""
        type="text"
        value=""
      />
    </label>
  </form>
</div>
`;
